test/unit/process/test_cuda_process_set.cpp

```diff
@@ -149,28 +159,39 @@ void testRandomSystemSubtractJacobianTerms(std::size_t n_cells, std::size_t n_re
     processes.push_back(micm::Process::create().reactants(reactants).products(products).phase(gas_phase));
   }
 
-  micm::ProcessSet cpu_set{ processes, state.variable_map_ };
-  micm::CudaProcessSet gpu_set{ processes, state.variable_map_ };
+  micm::ProcessSet cpu_set{ processes, cpu_state.variable_map_ };
+  micm::CudaProcessSet gpu_set{ processes, gpu_state.variable_map_ };
 
-  for (auto& elem : state.variables_.AsVector())
-    elem = get_double();
+  std::ranges::generate(cpu_state.variables_.AsVector(), get_double);
+  auto state_vars = cpu_state.variables_.AsVector();
+  gpu_state.variables_.AsVector().assign(state_vars.begin(), state_vars.end());
+  gpu_state.variables_.CopyToDevice();
 
-  MatrixPolicy<double> rate_constants{ n_cells, n_reactions };
-  for (auto& elem : rate_constants.AsVector())
-    elem = get_double();
+  CPUMatrixPolicy<double> cpu_rate_constants{ n_cells, n_reactions };
+  GPUDenseMatrixPolicy<double> gpu_rate_constants{ n_cells, n_reactions };
+  std::ranges::generate(cpu_rate_constants.AsVector(), get_double);
+  auto rate_vars = cpu_rate_constants.AsVector();
```

Comment:
```suggestion
  auto cpu_rate_vars = cpu_rate_constants.AsVector();
  std::ranges::generate(cpu_rate_vars, get_double);
```
Not a big deal but saves a call to `.AsVector()`?

Reply:
Thanks. Done.